scalableminds/webknossos
# One Version per Transaction

Adapts the transaction management when saving updates to an annotation. Now, the version number is not increased inside of a transaction. The goal is a cleaner history, especially for volume annotations. Every version that shows up in the version history/restore view, should have a consistent state across mags, and complete brush strokes.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- brush a high volume of data
- save
- look at version history and check that each version has consistent data

### TODOs:
  - [x] assume constant version for all groups of a transactions
  - [x] no longer assert increasing versions inside of a transaction
  - [x] instead assert increasing versions compared to the latest committed transaction
  - [x] also assert that the transactionGroupIndex is increasing monotonically when storing uncommitted update groups for a transaction
  - [x] transaction info is no longer optional in an update group, the frontend always sends this
  - [x] store uncommitted updates for more than 20 minutes, what if the user is offline for that time?
  - [x] send constant version number for all groups of a transaction

### Issues:
- fixes #6495

------
